templates/delivery/Jenkinsfile

```diff
@@ -3,8 +3,11 @@ pipeline {
     string(name: 'git_repository', description: 'The URL of the Git repository to clone when building the container image.')
     string(name: 'git_commit', description: 'The commit hash or branch name to checkout when building the container image.')
     string(name: 'git_credentials', description: 'The ID of the Jenkins credentials to use when cloning the Git repository.')
+    string(name: 'image_registry_auth_json', defaultValue: "{}" ,description: 'plain text JSON that can be used as docker config for authentication')
```

Comment:
`image_registry_auth_json` is now a template level `credential` type parameter in `template.yaml` instead of a string type pipeline level parameter in `Jenkinsfile` and is accessed using a `withCredentials` block.